model/tasks/SendAgsScoreTask.php

```diff
@@ -90,4 +106,38 @@ private function reportError(string $message): Report
 
         return Report::createError($message);
     }
+
+    private function retryTask(): void
+    {
+        if (!$this->isMaxRetryCountReached()) {
+            $this->increaseRetryCount();
+            $this->getQueueDispatcher()->createTask(new self, $this->params);
+            $this->logWarning('AGS score sending task has been retried');
+        }
+    }
+
+    public function increaseRetryCount(): void
+    {
+        $this->params[self::RETRY_COUNT]++;
+    }
+
+    public function isMaxRetryCountReached(): bool
+    {
+        return $this->params[self::RETRY_COUNT] >= $this->params[self::RETRY_MAX];
```

Comment:
Just in case the task would be created without a `RETRY_MAX`, I suggest returning a `true` in such a condition, not to run into a potential infinite retry.
```suggestion
        return ($this->params[self::RETRY_COUNT] ?? 0) >= ($this->params[self::RETRY_MAX] ?? 0);
```